src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineComputer.java

```diff
@@ -117,9 +123,18 @@ public HttpResponse doDoDelete() throws IOException {
         ComputeEngineInstance node = getNode();
         if (node != null) {
             try {
+                ComputeEngineCloud cloud = getCloud();
+
+                // Checks for failed jobs for this computer's node
+                if (cloud != null && node.getCreateSnapshot() && !this.getBuilds().failureOnly().isEmpty()) {
+                    LOGGER.log(Level.INFO, "Creating snapshot for node ... " + node.getNodeName());
+                    cloud.getClient().createSnapshot(cloud.projectId, node.zone, node.getNodeName());
+                }
+
                 node.terminate();
             } catch (InterruptedException ie) {
```

Comment:
What's the intended behavior when this exception is encountered? Swallowing an exception like this is a red flag.

Reply:
Not sure what you mean by swallowed exception. Do you mean the interruptedexception? That was already there.

Reply:
The InterruptedException is the one I'm referring to.

Reply:
@evandbrown care to opine?

Reply:
`git-blame` says that was me :) Definitely shouldn't be swallowing that exception (it's used: https://github.com/jenkinsci/jenkins/blob/371b9c134681e3e04f52a5e0bb39c747e6d44c45/core/src/main/java/hudson/model/AbstractItem.java#L666-L673). Just rethrow after after the log statement.

Reply:
Cool, thanks

Reply:
Also, just to point this out: if a snapshot fails the `node.terminate()` call will be skipped and the VM won't be terminated. I could see that being good (they wanted snapshots, the snapshots didn't happen, but the VM is still around so they can view logs) or bad (this VM that was supposed to be deleted is still running). Thoughts?

Reply:
O wait yikes, I remember I had tried to throw it before, but it doesn't agree with the same method in AbstractComputer
See: https://javadoc.jenkins.io/hudson/slaves/AbstractCloudComputer.html#doDoDelete--

doDoDelete there seems to be different since it takes in function arguments...

Reply:
O, is that because we will be unable to terminate because ComputeClient will throw an InterruptedException from timing out and we're just catching it in doDoDelete?

Hmm... yea it seems like it wouldn't be awful that the instance is still around to debug with, but does something like that usually frustrate users to have to manually delete the instance in the google cloud console?

Reply:
Alternatively, if InterruptedException is from creating the snapshot, I might just warn the user snapshot creation failed and not throw the exception so that deletion can still occur. Thoughts?

Reply:
@craigatgoogle thoughts?

Reply:
What's the behavior of the Jenkins environment if the exception is thrown vs. not?

Reply:
Won't disrupt the customer, but it will show up in logs that there was a deletion issue.